python_modules/dagster/dagster/core/instance/__init__.py

```diff
@@ -1050,15 +1054,27 @@ def create_reexecuted_run_from_failure(
             {
                 PARENT_RUN_ID_TAG: parent_run_id,
                 ROOT_RUN_ID_TAG: root_run_id,
-                RESUME_RETRY_TAG: "true",
             },
         )
+
         mode = cast(str, mode if mode is not None else parent_run.mode)
         run_config = run_config if run_config is not None else parent_run.run_config
 
-        step_keys_to_execute, known_state = get_retry_steps_from_parent_run(
-            self, parent_run=parent_run
-        )
+        if policy == ReexecutionPolicy.FROM_FAILURE:
+            check.invariant(
+                parent_run.status == PipelineRunStatus.FAILURE,
+                "Cannot reexecute from failure a run that is not failed",
+            )
+
+            step_keys_to_execute, known_state = get_retry_steps_from_parent_run(
+                self, parent_run=parent_run
+            )
+            tags[RESUME_RETRY_TAG] = "true"
+        elif policy == ReexecutionPolicy.ALL_STEPS:
+            step_keys_to_execute = None
+            known_state = None
```

Comment:
Pretty sure this should only be set if you're doing a partial execution?

Reply:
Yeah, I think that's right.